eth/internal/ethapi/api_test.go

```diff
@@ -0,0 +1,104 @@
+package ethapi_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/polymerdao/monomer/eth"
+	"github.com/polymerdao/monomer/eth/internal/ethapi"
+	"github.com/polymerdao/monomer/testutils"
+	"github.com/stretchr/testify/require"
+)
+
+// Constants for commonly used values
+const (
+	zeroString     = "0x0000000000000000000000000000000000000000000000000000000000000000"
+	someAddressHex = "0xabc"
+)
+
+// generateProofAPI creates a ProofAPI instance for testing
+func generateProofAPI(t *testing.T, blocksNumber int) *eth.ProofAPI {
+	t.Helper()
+	blockStore := testutils.NewLocalMemDB(t)
+	db := testutils.NewEthStateDB(t)
+
+	for i := 0; i < blocksNumber; i++ {
+		block := testutils.GenerateBlock(t)
+		require.NoError(t, blockStore.AppendBlock(block))
+		require.NoError(t, blockStore.UpdateLabels(block.Header.Hash, block.Header.Hash, block.Header.Hash))
+	}
+	return eth.NewProofAPI(db, blockStore)
+}
+
+func TestGetProof(t *testing.T) {
+	blockNumber := rpc.LatestBlockNumber
+	zeroHash := common.HexToHash(zeroString)
+	someAddress := common.HexToAddress(someAddressHex)
+	zeroBig := new(hexutil.Big)
+	require.NoError(t, zeroBig.UnmarshalText([]byte("0x0")))
+
+	testCases := []struct {
```

Comment:
Can you add test cases where the eth state db is populated so we can generate a proof? Right now we just have cases where no proof is generated.

For cases where we have a valid proof generated we'll want to verify that our proof was correctly generated as well and assert that. You may be able to use the optimism [VerifyProof](https://github.com/ethereum-optimism/optimism/blob/9936803afdc2848836eb1d69ecb68cef769b4288/op-node/withdrawals/proof.go#L80) func, if that doesn't work there's definitely helpers for this in the optimism or geth repos.

Reply:
@AnkushinDaniil I added a WIP commit to start the setup work for populating the eth state DB. Feel free to pick up where the commit leaves off (there's still a good bit of TODO's/refactoring) or I can take over this PR sometime in the near future

Reply:
[Add test case with populated eth state and blockstore](https://github.com/polymerdao/monomer/pull/230/commits/047c411dcd2f7d976b51be19ffdc0d014627d80a)